### DISCUSSION TITLE
Images with more than 3 channels

### DISCUSSION THREAD

**Discussion by ldv1** [francois-rozet/piqa#27] [Q&A] [answered]
Hi,

Some metrics (like SSIM) could be extended to 2D tensors (images) with more than 3 channels as the metrics are computed on a channel basis and then averaged over the channels. Is this the case in PIQA?

Best regards,
Laurent

**Comment by francois-rozet** [comment, marked as answer]
Hello @ldv1, 

Indeed, `SSIM`, `PSNR` and `TV` don't impose a specific number of channels as the computed quantities are averaged over the channels. However, all other metrics rely on the chromatic (e.g. RGB) decomposition of pixels.

For `SSIM`, as mentioned in the documentation (see https://francois-rozet.github.io/piqa/piqa.ssim.html#piqa.ssim.SSIM) and in a previous discussion (#22), the number of expected channels can be modified with the `n_channels` parameter.

**Reply by ldv1** [reply]
Hi,
Sorry, this is something I overlooked.
Since this might be of interest to a wide audience, this information could be put in the main README.
Big thanks for your prompt answer.
